haystack/preview/components/preprocessors/text_document_cleaner.py

```diff
@@ -0,0 +1,205 @@
+import re
+from copy import deepcopy
+from functools import partial, reduce
+from itertools import chain
+from typing import Any, Dict, Generator, List, Optional, Set
+
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+@component
+class TextDocumentCleaner:
```

Comment:
The naming is not ideal, I agree. TextDocumentCleaner would be consistent with TextDocumentSplitter. I guess we should rename both to DocumentCleaner and DocumentSplitter. The idea for TextDocumentSplitter probably came up because we might need to split TableDocuments in a different way, etc.

Reply:
I'll rename to DocumentCleaner now. The DocumentSplitter renaming will be a breaking change and should be done in a separate PR.